packages/features/webhooks/lib/service/WebhookNotificationHandler.ts

```diff
@@ -0,0 +1,63 @@
+import logger from "@calcom/lib/logger";
+
+import type { WebhookEventDTO } from "../dto/types";
+import type { WebhookPayloadFactory } from "../factory/WebhookPayloadFactory";
+import type { IWebhookService } from "../interface/services";
+import type { IWebhookNotificationHandler } from "../interface/webhook";
+
+const log = logger.getSubLogger({ prefix: ["[WebhookNotificationHandler]"] });
+
+export class WebhookNotificationHandler implements IWebhookNotificationHandler {
+  constructor(
+    private readonly webhookService: IWebhookService,
+    private readonly webhookPayloadFactory: WebhookPayloadFactory
+  ) {}
+
+  async handleNotification(dto: WebhookEventDTO, isDryRun = false): Promise<void> {
+    const trigger = dto.triggerEvent;
+
+    try {
+      if (isDryRun) {
+        log.debug(`Dry run mode - skipping webhook notification for: ${trigger}`);
+        return;
+      }
+
+      const queryParams = {
```

Comment:
nit: maybe rename it to subscriptionParams. I got it mixed up with the browser query params for some time 😄